core/src/index.ts

```diff
@@ -58,3 +59,20 @@ export { waitForLock } from "./modules/locks";
 export { Errors } from "./modules/errors";
 export * from "./modules/cache";
 export * from "./modules/tableSpeculation";
+
+// ensure config types are loaded
+import "./config/api";
+import "./config/batchSize";
+import "./config/environment";
+import "./config/errors";
+import "./config/logger";
+import "./config/next";
+import "./config/pluginInjection";
+import "./config/plugins";
+import "./config/redis";
+import "./config/routes";
+import "./config/sequelize";
+import "./config/tasks";
+import "./config/telemetry";
+import "./config/web";
+import "./config/websocket";
```

Comment:
This was an interesting trick - to force consumers (aka plugins) to be aware of the config types, they need to source the module overrides in the config files.  Adding them here in the main export file makes that happen!